core/arch/arm/kernel/thread.c

```diff
@@ -1285,24 +1212,102 @@ void thread_rpc_alloc_arg(size_t size, paddr_t *arg, uint64_t *cookie)
 	*cookie = co;
 }
 
+static bool get_rpc_arg(uint32_t cmd, size_t num_params,
+			struct optee_msg_arg **arg_ret, uint64_t *carg_ret,
+			struct optee_msg_param **params_ret)
+{
+	struct thread_ctx *thr = threads + thread_get_id();
+	struct optee_msg_arg *arg = thr->rpc_arg;
+
+	if (num_params > THREAD_RPC_MAX_NUM_PARAMS)
+		return false;
+
+	if (!arg) {
+		paddr_t p;
+		uint64_t c;
+
+		thread_rpc_alloc_arg(
+			OPTEE_MSG_GET_ARG_SIZE(THREAD_RPC_MAX_NUM_PARAMS),
+			&p, &c);
+		if (!p || !ALIGNMENT_IS_OK(p, struct optee_msg_arg))
+			return false;
```

Comment:
Missing free of the allocated memory when it does not fulfil the alignment constraint.